operator/pkg/ciliumidentity/controller.go

```diff
@@ -93,6 +93,10 @@ func registerController(p params) {
 		return
 	}
 
+	if !p.SharedCfg.EnableNetworkPolicySystem {
```

Comment:
nit: could collapse this with the above and `cmp.Or`?

Reply:
Done.